architecture/2024-08-controller-component-threat-model.md

```diff
@@ -0,0 +1,241 @@
+
+#### Compromised TLS Certificates and private keys of the Validating Webhook
+
+Risk: If the TLS certificates and the private keys used by the validating webhook are compromised, an attacker could impersonate the webhook server, intercepting the requests from the Kubernetes API Server or altering the responses.
+
+Impact: The validating webhook functions should return an approval or a validation error to the caller. The impersonater can return incorrect responses to the Kubernetes API Server which could affect the lifecycle of resources.
+
+Risk: If the TLS certificates and the private keys used by the validating webhook are compromised, an attacker could enable unauthorized clients to communicate with the webhook.
+
+Impact: Unauthorized clients wouldn't have any affect in the lifecycle of resources because, even if the webhook returns an approval or a rejection to the unauthorized client, the unauthorized client should also have access to the Kubernetes API Server to further trigger changes in the resource.
+
+Mitigation:
+
+1. **mTLS (Mutual TLS)**: Implement mutual TLS to ensure that both the client (Kubernetes API Server) and the server (Validating Webhook) authenticate each other. This helps in verifying the authenticity of both parties and prevents unauthorized access. Note that mTLS is enabled for the communication between the Validating Webhook and the Kubernetes API Server in Radius as of now.
+2. **Network Policies**: Implement necessary Network Policies to ensure that communication between components is restricted to authorized components only. In this case, the Validating Webhook and the Kubernetes API Server should only accept requests from each other.
+3. **Strong Certificate Management**: Implement strong certificate management practices, including regular rotation of certificates, using short-lived certificates, and monitoring for certificate anomalies.
+4. **Secure Storage**: Store TLS certificates securely using secrets management solutions to prevent unauthorized access. The certificates and the private keys are being kept in a Secret object on Kubernetes.
+
+#### Compromised TLS Certificates and private keys of the Kubernetes API Server
+
+Risk: If the TLS certificates and private keys used by the Kubernetes API Server are compromised, an attacker could impersonate the Kubernetes API Server. This would allow the attacker to intercept, alter, or forge communication between the API Server and other components, including the Validating Webhook and Reconcilers. This could lead to unauthorized access, data breaches, and manipulation of cluster operations.
+
+Impact:
+
+1. The attacker could gain unauthorized access to sensitive data including the data it can get from the etcd if proper policies are not in place.
+2. The attacker can manipulate the operations of the cluster, causing unauthorized changes.
```

Comment:
Are these threats related to our components?

Reply:
Actually, no. I am removing the threat that includes the compromise of certs and private key of Kubernetes API Server.